### PR TITLE
Display management in gui vm

### PR DESCRIPTION
**Pull request by gngram**
## Description of changes
- Turn off the display when system is suspended (Lid closed) and turn on when it resumes(Lid Open).
- Host notifies admin vm for the events and admin vm starts and stops display-resume and display suspend services based on event on gui vm.

More details are available on [Confluence](https://ssrc.atlassian.net/wiki/spaces/GA/pages/1428357127/Power+and+Display+Management+in+Laptops).

## Checklist for things done

- [ ] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [ ] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `make-checks` and it passes
- [x] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status
- [ ] Change requires full re-installation
- [ ] Change can be updated with `nixos-rebuild ... switch`

## Instructions for Testing

- [x] List all targets that this applies to: Lenovo X1 `x86_64`
- [x] Is this a new feature
  - [x] List the test steps to verify:
  - [x] Close laptop lid and wait for 3-4 seconds and open the lid and check if display is suspended and restored.
  - [x] Also check gui-vm and host-vm journalctl log for error. There should not be any error regarding display suspend and resume.
- [ ] If it is an improvement how does it impact existing functionality?

### CODE REVIEW

File: modules/givc/adminvm.nix [hunk 1]
```diff
@@ -23,6 +23,8 @@ in
         "givc-net-vm.service"
         "givc-gui-vm.service"
         "givc-audio-vm.service"
+        "display-suspend.service"
+        "display-resume.service"
```
Comment (mbssrc) [inline modules/givc/adminvm.nix:27]: These aren't necessary to include here

File: modules/givc/host.nix [hunk 2]
```diff
@@ -31,8 +31,11 @@ in
         inherit addr;
         port = "9000";
         services = [
+          "microvm@admin-vm.service"
```
Comment (mbssrc) [inline modules/givc/host.nix:34]: This is already included

File: modules/givc/host.nix [hunk 3]
```diff
@@ -31,8 +31,11 @@ in
...
           "reboot.target"
           "poweroff.target"
+          "sleep.target"
+          "suspend.target"
```
Comment (mbssrc) [inline modules/givc/host.nix:38]: These targets are currently not used. We could add a conditional invocation of these targets if the requests originates in gui-vm

File: modules/microvm/virtualization/microvm/guivm.nix [hunk 4]
```diff
@@ -89,6 +89,35 @@ let
...
+              Type = "oneshot";
+              ExecStart = ''
+                ${pkgs.wlopm}/bin/wlopm --on '*'  
+              '';
+              Environment = ''
+                WAYLAND_DISPLAY=/run/user/1000/wayland-0 
+              '';
+            };
+          };
+
```
Comment (mbssrc) [inline modules/microvm/virtualization/microvm/guivm.nix:120]: Is the display-suspend service executed before host suspends/freezes the `microvm@gui-vm.service`?

**Comment by mbssrc** [comment]
This bumps GIVC, so we need general checking of GIVC functionality as well. This also impacts #710 as the bump is newer.

File: modules/givc/adminvm.nix (hunk 1 above)
Comment (gangaram-tii) [reply modules/givc/adminvm.nix]: Removed in latest push.

File: modules/givc/host.nix (hunk 2 above)
Comment (gangaram-tii) [reply modules/givc/host.nix]: ok, removed.

File: modules/microvm/virtualization/microvm/guivm.nix (hunk 4 above)
Comment (gangaram-tii) [reply modules/microvm/virtualization/microvm/guivm.nix:120]: Yes, display suspend is called before pre-sleep actions.

**Review by mbssrc — APPROVED** [review]

**Comment by johannarautanen** [comment]
**Tested with lenovo X1**. flash-script used. Commit: b4b747bb7a783efcbbacf6a6c39762239dbca7d0

Working:
-  Occasionally when you close laptop lid and wait for 3-4 seconds display is suspended and you open the lid it is resumed. you can opened a web page before closing the lid. 
- all other ghaf apps can be launched
- ci-test automation cases ok

Issues:
- most of the time is took about 20 seconds before display is suspended, when you open the lid it is resumed. You have save test editor file before closing the lid.
- There was error in gui-vm and ghaf-host:
[guivm.txt](https://github.com/user-attachments/files/17521091/guivm.txt)
[ghafhost.txt](https://github.com/user-attachments/files/17521094/ghafhost.txt)

Screenshot from 2024-10-25 14-11-47
Screenshot from 2024-10-25 14-12-22

**Comment by gangaram-tii** [comment]
Don't merge this... will update when it is ready for merging.

**Comment by gangaram-tii** [comment]
@johannarautanen Please retest this PR once it passes all the checks.

**Pull request merged**